src/platform/CMakeLists.txt

```diff
@@ -48,22 +57,84 @@ else()
     message(FATAL_ERROR "TLS Provider not configured")
 endif()
 
+
 add_library(platform STATIC ${SOURCES})
 
-if("${CX_PLATFORM}" STREQUAL "windows")
+if ("${CX_PLATFORM}" STREQUAL "windows")
     target_link_libraries(
         platform
         PUBLIC
-        inc
         wbemuuid)
-else()
-    target_link_libraries(platform PUBLIC inc)
-endif()
-
-if("${CX_PLATFORM}" STREQUAL "windows")
     target_link_libraries(platform PUBLIC winmm)
+elseif(LINUX_XDP_ENABLED)
+    # XDP Tools and libbpf paths
+    set(XDP_TOOLS_DIR ${PROJECT_SOURCE_DIR}/submodules/xdp-tools)
+    set(LIBXDP_DIR ${XDP_TOOLS_DIR}/lib/libxdp)
+    set(LIBBPF_DIR ${XDP_TOOLS_DIR}/lib/libbpf/src)
+    set(LIBXDP_PATH ${LIBXDP_DIR}/libxdp.so)
+    set(LIBBPF_PATH ${LIBBPF_DIR}/libbpf.so)
+
+    # Custom command to build XDP Tools
+    add_custom_command(
+        OUTPUT ${LIBXDP_PATH} ${LIBBPF_PATH} ${QUIC_OUTPUT_DIR}/xdp-dispatcher.o
+        COMMAND ./configure
+        COMMAND make
+        # Experimental!
+        # xdp-dispatcher.o need to be placed at
+        # - "./" (libxdp Debug build)
+        # - "/usr/lib/bpf"
+        # - Specified by LIBXDP_OBJECT_PATH
+        COMMAND cp ${XDP_TOOLS_DIR}/lib/libxdp/xdp-dispatcher.o ${QUIC_OUTPUT_DIR}/
+        WORKING_DIRECTORY ${XDP_TOOLS_DIR}
+    )
+
+    # Add custom target and make platform depend on it
+    add_custom_target(xdp_tools DEPENDS ${LIBXDP_PATH} ${LIBBPF_PATH})
+    add_dependencies(platform xdp_tools)
+
+    # Include directories for libxdp.so and libbpf.so
+    target_include_directories(platform PRIVATE ${XDP_TOOLS_DIR}/headers/xdp)
+    target_include_directories(platform PRIVATE ${XDP_TOOLS_DIR}/lib/libbpf/src)
+    target_include_directories(platform PRIVATE ${XDP_TOOLS_DIR}/lib/libbpf/src/root/usr/include) # TODO: use installed path
+
+    # building XDP program
+    add_custom_command(
+        OUTPUT ${PROJECT_SOURCE_DIR}/datapath_raw_xdp_kern.o
+        OUTPUT ${QUIC_OUTPUT_DIR}/datapath_raw_xdp_kern.o
+        COMMAND clang -O2 -g -target bpf -c ${PROJECT_SOURCE_DIR}/src/platform/datapath_raw_xdp_linux_kern.c -o ${QUIC_OUTPUT_DIR}/datapath_raw_xdp_kern.o -I${PROJECT_SOURCE_DIR}/submodules/xdp-tools/lib -I/usr/include/x86_64-linux-gnu
+        DEPENDS ${PROJECT_SOURCE_DIR}/src/platform/datapath_raw_xdp_linux_kern.c
+    )
+
+    add_custom_target(xdp_prog ALL
+        DEPENDS ${QUIC_OUTPUT_DIR}/datapath_raw_xdp_kern.o)
+    add_dependencies(platform xdp_prog)
+
+    find_library(NL_LIB nl-3)
+    find_library(NL_ROUTE_LIB nl-route-3)
+    if (BUILD_SHARED_LIBS)
+        add_custom_target(
+            CopyLibsAndLinks ALL
+            COMMAND cp -P ${LIBXDP_DIR}/*.so* ${QUIC_OUTPUT_DIR}/
```

Comment:
Because `cargo build --all` sets `QUIC_OUTPUT_DIR` to '../lib` and maybe it is generated at the end of build, the directory doesn't find

Reply:
Or these libxdp and libbpf artifact should be .a and statically linked to libmsquic.so? (how?)